server/src/main/java/org/elasticsearch/cluster/service/MasterService.java

```diff
@@ -702,6 +702,7 @@ private static class TaskAckListener {
             ThreadPool threadPool
         ) {
             this.contextPreservingAckListener = contextPreservingAckListener;
+            this.ackTimeout = Objects.requireNonNull(contextPreservingAckListener.ackTimeout());
```

Comment:
NB reading this in the constructor so that the whole cluster state update fails before starting with a NPE rather than throwing in `onCommit`.